blog-posts/sls-security-headers/sls-security-headers.md

```diff
@@ -0,0 +1,96 @@
+| [X-Content-Type-Options](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Content-Type-Options) | nosniff | Prevents older browsers from sniffing the content of files sent by the server rather than trusting the Content-Type header. This prevents the detection and execution of malicious Javascript in an image uploaded by an attacker and then retrieved by a victim. |
+| [X-Frame-Options](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options) | SAMEORIGIN | Prevents the display of the site in an iframe, and thus protects against click-jacking. |
+| [X-XSS-Protection](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection) | 1; mode=block | Prevents rendering of the page if an XSS attack is detected. |
+| [Referrer-Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referrer-Policy) | strict-origin-when-cross-origin | Prevents sending sensitive information (origin, path and query string) to another website or when the protocol is downgraded (HTTPS → HTTP) |
+
+An extensive list can be found on the Mozilla Developer Network (MDN).
+
+[https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers#security](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers#security)
+
+HTTP security headers are not *per se* good coding practices, but rather good running strategies. As a result, they fall under the responsibility of ops teams.
+
+Let’s digress a little bit to mention the Cross Origin Resource Sharing (CORS) headers. As a web application developer, you certainly already have come across them. Otherwise, believe me, you will encounter them soon or later. 😉
+
+CORS headers are different because:
+
+- weaken a security feature of the browser of your users, [reformulate]
+- are a pain to deal with in your development journey. 😅
+
+These headers do not impact the application itself. Thus adding these can be done at a very low cost.
+
+
+# Demystifying HTTP security headers on AWS
+
+AWS mentions [several methods](https://aws.amazon.com/websites/) to deploy your website.
+
+## S3
+
+Amazon Simple Storage Service (S3) offers the option to host static websites. However, with this method, configuring HTTPS, and all the more so HTTP security headers, is not possible.
+
+## CloudFront
+
+On the contrary, **HTTP security headers can be very easily enabled on CloudFront**. If this is your first time dealing with CloudFront:
+
+- don’t worry, this is actually pretty easy and cheap,
+- brace yourself, CloudFront’s performance is mind blowing,
+
+Let’s dive in!
+
+The CloudFront service works by creating _distributions_. These describe how your content will be delivered to your users.
+
+Head over to the CloudFront service in the console.
+
+Create of modify the distribution of your choice. Here, let’s assume the origin is S3 (alternatively this could be your own HTTP server).
+
+![edit-behavior](./assets/edit-behavior.png)
```

Comment:
Screenshots with the default light theme would be better since you are using a browser extension to have a dark theme, people will be kind of lost when seeing this.